src/citrine/resources/material_template.py

```diff
@@ -115,3 +116,48 @@ class MaterialTemplateCollection(DataConceptsCollection[MaterialTemplate]):
     def get_type(cls) -> Type[MaterialTemplate]:
         """Return the resource type in the collection."""
         return MaterialTemplate
+
+    def get_specs(self, scope: str, id: str, per_page: int = 20) -> Iterator[dict]:
+        """
+        [ALPHA] Get all material specs associated with a material template.
+
+        The material template is specified by its scope and id.
+
+        :param scope: The scope used to locate the material template.
+        :param id: The unique id corresponding to `scope`.
+            The lookup will be most efficient if you use the Citrine ID (scope='id')
+            of the material template.
+        :param per_page: The number of results to return per page.
+        :return: A search result of material specs
+        """
+        path = self._get_path(ignore_dataset=True) + "/" + scope + "/" + id + "/material-specs"
+        return self.session.cursor_paged_resource(self.session.get_resource,
+                                                  path,
+                                                  per_page=per_page,
+                                                  version="v1")
+
+    # Retrieve all material runs associated with a material template
+    def get_runs(self,
+                 template_scope: str,
+                 template_id: str,
+                 spec_collection: MaterialSpecCollection,
```

Comment:
I think it would be preferable to not make our users pass in `dataset.material_specs` / `project.material_specs` here -- it should be possible to just make the collection in the body like we do from the project https://github.com/CitrineInformatics/citrine-python/blob/30c6313329299fd38cf88d48bd0fd9e897d5f2ad/src/citrine/resources/project.py#L183